### PR TITLE
Send gdpr data directly

### LINKED ISSUE #388: Directly send gdpr data
**Problem:** The current gdpr implementation kinda works. However there are some reports we cant send since the user doesnt share a server with us anymore.

For some reason in the past we decided to collect reports and process them in a batch request. Probably to avoid unexpected load on the system. However this only moves the load to a single point causing load spikes. We also probably did this to avoid abusing the system to overload our database by too many request. But thats also not an issue since you can only request your data once a month.

**Solution:**
Directly send the data instead of queuing it.

- [ ] Directly send the data in the gdpr command
- [ ] Still log the gdpr send and received time in case of an unexpected failure
- [ ] Remove the scheduler for the reports

### PR DESCRIPTION
**Pull request by RainbowDashLabs**
**Checklist**
- [x] I made changes to commands
  - [x] I fully localised the command
  - [x] I fully checked the commands functionality

- [x] I added new localisation codes
  - [x] I fully translated it for all languages
  - [x] I sorted properties alphabetically

- [ ] I made changes to the database
  - [ ] I added my changed to a patch file
  - [ ] I made sure my database was up to date
  - [ ] I checked that the migration works

- [x] I made changes to internal structure 

**Short Description**

Send gdpr directly.

**Detailed Description**

GDPR data is now send directly after requesting it.
If the send failed it will be queued by the old known mechanism.
After every failed attempt the next send will be delayed by one hour up to 48 hours in total. This will reduce the numbers of errors and failed attempts by a lot hopefully

Closes #388

### CODE REVIEW

File: src/main/java/de/chojo/repbot/dao/access/gdpr/GdprUser.java [hunk 1]
```diff
@@ -160,9 +160,13 @@ public boolean sendData(ShardManager shardManager) {
                     .addFiles(FileUpload.fromData(tempFile.toFile()))
                     .complete();
         } catch (RuntimeException e) {
-            log.warn("Could not send gdpr data to user {}. File sending failed.", userId, e);
+            log.warn("Could not send gdpr data to user {}. File sending failed.", userId(), e);
```
Comment (Goldmensch) [inline src/main/java/de/chojo/repbot/dao/access/gdpr/GdprUser.java:163]: Do you really want to include the exception here? Also perhaps you should insert more information like the errorcode in the log message.

File: src/main/java/de/chojo/repbot/dao/access/Gdpr.java [hunk 2]
```diff
@@ -45,18 +47,26 @@ public void cleanupRequests() {
...
+        return new GdprUser(this, user);
     }
 
     /**
      * Get a list of {@link GdprUser}s which have requested their data.
      *
      * @return list of users
      */
-    public List<GdprUser> getReportRequests() {
+    public List<GdprUser> getReportRequests(ShardManager shardManager) {
```
Comment (Goldmensch) [inline src/main/java/de/chojo/repbot/dao/access/Gdpr.java:58]: Adjust javadoc to reflect the new time condition

**Review by Goldmensch — CHANGES_REQUESTED** [review]

**Comment by Goldmensch** [comment]
I also can't find the code, which is responsibly to delay the next send.

File: src/main/java/de/chojo/repbot/dao/access/gdpr/GdprUser.java (hunk 1 above)
Comment (RainbowDashLabs) [reply src/main/java/de/chojo/repbot/dao/access/gdpr/GdprUser.java:163]: Yeah message of the error might suffice.

**Comment by RainbowDashLabs** [comment]
> I also can't find the code, which is responsibly to delay the next send.

It is here
https://github.com/RainbowDashLabs/reputation-bot/pull/420/files#diff-b6c9193910c4221ae654a80842287949bdcb1ca6c5b57a0b0a7efa5f13f16a54R67

**Comment by Goldmensch** [comment]
But this method isn't called when a user requests it's data. Or I'm missing something?

**Review by Goldmensch — APPROVED** [review]

**Pull request merged**